File: app/src/main/java/com/jameskbride/eventbusmvpdemo/network/NetworkErrorViewFragment.java
package com.jameskbride.eventbusmvpdemo.network;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.DialogFragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.jameskbride.eventbusmvpdemo.EventBusMVPDemoApplication;

import javax.inject.Inject;

public class NetworkErrorViewFragment extends DialogFragment {

    public static final String NETWORK_REQUEST = "NETWORK_REQUEST";

    @Inject
    NetworkErrorViewFragmentImpl delegate;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        ((EventBusMVPDemoApplication)getActivity().getApplication()).getApplicationComponent().inject(this);

        return delegate.onCreateView(inflater, container, savedInstanceState, this);
    }

    @Override
    public void onResume() {
        super.onResume();
        delegate.onResume(this);
    }

    @Override
    public void onPause() {
        super.onPause();

        delegate.onPause(this);
    }
}

File: app/src/main/java/com/jameskbride/eventbusmvpdemo/network/NetworkErrorViewFragment.kt
package com.jameskbride.eventbusmvpdemo.network

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.DialogFragment
import com.jameskbride.eventbusmvpdemo.EventBusMVPDemoApplication
import javax.inject.Inject


class NetworkErrorViewFragment : DialogFragment() {

    @Inject
    lateinit var impl: NetworkErrorViewFragmentImpl

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        EventBusMVPDemoApplication.applicationComponent.inject(this)

        val view = impl.onCreateView(inflater, container, savedInstanceState, this)

        return view
    }

    override fun onResume() {
        super.onResume()
        impl.onResume(this)
    }

    override fun onPause() {
        super.onPause()

        impl.onPause(this)
    }

    companion object {
        val NETWORK_REQUEST = "NETWORK_REQUEST"
    }
}